test/react/hooks/use-messages.integration.test.tsx

```diff
@@ -188,4 +188,207 @@ describe('useMessages', () => {
     expect(results.items.find((item) => item.text === 'I have the high ground')).toBeDefined();
     expect(results.items.find((item) => item.text === 'You underestimate my power')).toBeDefined();
   }, 10000);
+
+  it('should reset getPreviousMessages if the listener becomes undefined then redefined', async () => {
+    const chatClient = newChatClient() as unknown as ChatClient;
+
+    // create a second room instance so we can send messages from it
+    const roomId = randomRoomId();
+    const room = chatClient.rooms.get(roomId, RoomOptionsDefaults);
+
+    // send a few messages before the first room has subscribed
+    await room.messages.send({ text: 'The force is strong with this one' });
+    await room.messages.send({ text: 'I have the high ground' });
+    await room.messages.send({ text: 'You underestimate my power' });
+
+    // Wait 1 second to make sure all messages are received
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
Should we avoid arbitrary waits? Also, do we need it? I thought that because the realtime serial we get from attachments must always be after sending of the last few messages (since this happens serially in channel) we wouldn't need to wait beyond sending of the messages above? :)

Reply:
We needed a small wait to ensure messages propagate around realtime - not waiting would sometimes mean the re-render would happen a tad too quick :)